fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/rules/SimplifyRange.java

```diff
@@ -0,0 +1,390 @@
+            // only handle `NumericType`
+            if (right.isLiteral() && right.getDataType().isNumericType()) {
+                return ValueDesc.range((ComparisonPredicate) rewrite);
+            }
+            return ValueDesc.EMPTY;
+        }
+
+        @Override
+        public ValueDesc visitGreaterThan(GreaterThan greaterThan, Void context) {
+            return buildRange(greaterThan);
+        }
+
+        @Override
+        public ValueDesc visitGreaterThanEqual(GreaterThanEqual greaterThanEqual, Void context) {
+            return buildRange(greaterThanEqual);
+        }
+
+        @Override
+        public ValueDesc visitLessThan(LessThan lessThan, Void context) {
+            return buildRange(lessThan);
+        }
+
+        @Override
+        public ValueDesc visitLessThanEqual(LessThanEqual lessThanEqual, Void context) {
+            return buildRange(lessThanEqual);
+        }
+
+        @Override
+        public ValueDesc visitEqualTo(EqualTo equalTo, Void context) {
+            return buildRange(equalTo);
+        }
+
+        @Override
+        public ValueDesc visitInPredicate(InPredicate inPredicate, Void context) {
+            // only handle `NumericType`
+            if (ExpressionUtils.isAllLiteral(inPredicate.getOptions())
+                    && ExpressionUtils.matchNumericType(inPredicate.getOptions())) {
+                return ValueDesc.discrete(inPredicate);
+            }
+            return ValueDesc.EMPTY;
+        }
+
+        @Override
+        public ValueDesc visitAnd(And and, Void context) {
+            List<Expression> result = simplify(ExpressionUtils.extractConjunction(and), ValueDesc::intersect);
+            return ResultValue.of(ExpressionUtils.and(result));
+        }
+
+        @Override
+        public ValueDesc visitOr(Or or, Void context) {
+            List<Expression> result = simplify(ExpressionUtils.extractDisjunction(or), ValueDesc::union);
+            return ResultValue.of(ExpressionUtils.or(result));
+        }
+
+        private List<Expression> simplify(List<Expression> predicates, BinaryOperator<ValueDesc> op) {
+            List<Expression> result = Lists.newArrayList();
+            List<ValueDesc> valueDescList = Lists.newArrayList();
+
+            for (Expression predicate : predicates) {
+                ValueDesc value = predicate.accept(this, null);
+                // can not build `ValueDesc`, so does not handle `predicate`, skip it.
+                if (value.equals(ValueDesc.EMPTY)) {
+                    result.add(predicate);
+                } else if (value instanceof ResultValue) {
+                    // With simplified expression, it is possible to build `ValueDesc` as well
+                    ValueDesc o = value.expr.accept(this, null);
+                    if (o instanceof ResultValue) {
+                        result.add(o.expr);
+                    } else {
+                        valueDescList.add(o);
+                    }
```

Comment:
Why simplify `resultValue.expr` again?
I think you can direct add to the result list: 
```java
else if (value instanceof ResultValue) {
  result.add(o.expr);
}
```